recovery/src/main/java/com/kin/ecosystem/recovery/backup/view/BackupActivity.java

```diff
@@ -0,0 +1,88 @@
+package com.kin.ecosystem.recovery.backup.view;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.view.View;
+import android.view.View.OnClickListener;
+import com.kin.ecosystem.recovery.R;
+import com.kin.ecosystem.recovery.backup.presenter.BackupPresenter;
+import com.kin.ecosystem.recovery.backup.presenter.BackupPresenterImpl;
+import com.kin.ecosystem.recovery.base.BaseToolbarActivity;
+
+public class BackupActivity extends BaseToolbarActivity implements BackupView {
+
+	private BackupPresenter backupPresenter;
+
+	@Override
+	protected int getContentLayout() {
+		return R.layout.kinrecovery_frgment_activity;
+	}
+
+	@Override
+	protected void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setNavigationClickListener(new OnClickListener() {
+			@Override
+			public void onClick(View v) {
+				backupPresenter.onBackClicked();
+			}
+		});
+		backupPresenter = new BackupPresenterImpl();
+		backupPresenter.onAttach(this);
+	}
+
+	public void startBackupFlow() {
+		setToolbarColor(R.color.kinrecovery_bluePrimary);
+		setNavigationIcon(R.drawable.kinrecovery_ic_back);
+		setToolbarTitle(EMPTY_TITLE);
+		BackupInfoFragment backupFragment = (BackupInfoFragment) getSupportFragmentManager()
+			.findFragmentByTag(BackupInfoFragment.class.getSimpleName());
+
+		if (backupFragment == null) {
+			backupFragment = BackupInfoFragment.newInstance(backupPresenter);
```

Comment:
this is could be dangerous: the current   backupPresenter could be diffrent then then the one used in the current BackupInfoFragment